org/postgresql/test/jdbc2/JdbcTimestampTest.java

```diff
@@ -0,0 +1,214 @@
+*
+*
+*-------------------------------------------------------------------------
+*/
+package org.postgresql.test.jdbc2;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Properties;
+import java.util.TimeZone;
+
+import junit.framework.TestCase;
+
+import org.postgresql.PGProperty;
+import org.postgresql.test.TestUtil;
+import org.postgresql.util.PGTimestamp;
+
+/**
+ * Tests {@link java.sql.Timestamp and PGTimestamp} in various scenarios including setTimestamp, setObject for both
+ * <code>timestamp with time zone</code> and <code>timestamp without time zone</code> data types.
+ */
+public class JdbcTimestampTest extends TestCase
+{
+    /** The name of the test table. */
+    private static final String TEST_TABLE = "testjdbctimestamp";
+
+    /** The database connection. */
+    private Connection con;
+
+    public JdbcTimestampTest(final String name)
+    {
+        super(name);
+    }
+
+    protected void setUp() throws Exception
+    {
+        Properties props = new Properties();
+        props.setProperty(PGProperty.STRICT_TIMESTAMP.getName(), "true");
+        con = TestUtil.openDB(props);
+        TestUtil.createTable(con, TEST_TABLE, "ts timestamp, tz timestamp with time zone");
+    }
+
+    protected void tearDown() throws Exception
+    {
+        TestUtil.dropTable(con, TEST_TABLE);
+        TestUtil.closeDB(con);
+    }
+
+    public void testParameterIsNull() throws SQLException{
+        long now = System.currentTimeMillis();
+        PreparedStatement insertStmt = con.prepareStatement("INSERT INTO " + TEST_TABLE + " VALUES (?, ?)");
+        insertStmt.setTimestamp(1, new java.sql.Timestamp(now++));
+        insertStmt.setTimestamp(2, new java.sql.Timestamp(now++));
+        assertEquals(1, insertStmt.executeUpdate());
+        insertStmt.setTimestamp(1, new java.sql.Timestamp(now++));
+        insertStmt.setTimestamp(2, new java.sql.Timestamp(now++));
+        assertEquals(1, insertStmt.executeUpdate());
+        insertStmt.setTimestamp(1, new java.sql.Timestamp(now++));
+        insertStmt.setTimestamp(2, new java.sql.Timestamp(now++));
+        assertEquals(1, insertStmt.executeUpdate());
+        insertStmt.close();
+        PreparedStatement selectStmt = con.prepareStatement("Select * from " + TEST_TABLE + " Where ts = ? or ? is null");
+        selectStmt.setTimestamp(1, new java.sql.Timestamp(now++));
+        selectStmt.setTimestamp(2, new java.sql.Timestamp(now++));
+        ResultSet rs1 = selectStmt.executeQuery();
+        assertFalse(rs1.next());
+        rs1.close();
+        
+        selectStmt.setTimestamp(1, new java.sql.Timestamp(now++));
+        selectStmt.setTimestamp(2, null);
+        ResultSet rs2 = selectStmt.executeQuery();
+        assertTrue(rs2.next());
```

Comment:
Either avoid `assertTrue` in such not really important cases, or ensure you have descriptive messages: https://github.com/pgjdbc/pgjdbc#test